contentgrid-automations-rest/src/main/java/com/contentgrid/automations/rest/AutomationModelVisitor.java

```diff
@@ -0,0 +1,141 @@
+package com.contentgrid.automations.rest;
+
+import com.contentgrid.automations.rest.AutomationsModel.AutomationModel;
+import com.contentgrid.thunx.predicates.model.FunctionExpression;
+import com.contentgrid.thunx.predicates.model.Scalar;
+import com.contentgrid.thunx.predicates.model.SymbolicReference;
+import com.contentgrid.thunx.predicates.model.SymbolicReference.PathElement;
+import com.contentgrid.thunx.predicates.model.SymbolicReference.PathElementVisitor;
+import com.contentgrid.thunx.predicates.model.ThunkExpression;
+import com.contentgrid.thunx.predicates.model.ThunkExpressionVisitor;
+import com.contentgrid.thunx.predicates.model.Variable;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+class AutomationModelVisitor implements ThunkExpressionVisitor<Scalar<?>, AutomationModel> {
+
+    private static final String SUBJECT_NAME = "automation";
+
+    @Override
+    public Scalar<?> visit(Scalar<?> scalar, AutomationModel context) {
+        return scalar;
+    }
+
+    @Override
+    public Scalar<?> visit(FunctionExpression<?> functionExpression, AutomationModel context) {
+        var terms = functionExpression.getTerms().stream()
+                .map(expr -> expr.accept(this, context))
+                .toList();
+
+        return switch (functionExpression.getOperator()) {
+            case EQUALS -> {
+                assertTwoTerms(terms);
+                yield Scalar.of(terms.get(0).equals(terms.get(1)));
+            }
+            case NOT_EQUAL_TO -> {
+                assertTwoTerms(terms);
+                yield Scalar.of(!terms.get(0).equals(terms.get(1)));
+            }
+            case AND -> {
+                yield Scalar.of(terms.stream()
+                        .map(scalar -> scalar.assertResultType(Boolean.class))
+                        .map(ThunkExpression::maybeValue)
+                        .flatMap(Optional::stream)
+                        .map(Boolean.class::cast)
+                        .reduce(true, (left, right) -> left && right));
+            }
+            case OR -> {
```

Comment:
Should we assert that there's _at least_ one term for this? Otherwise the reduce will just fall back to false and you won't actually see an exception telling you there's something wrong with the policy. Same for `case AND`.